zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -41,3 +50,59 @@ impl TryFrom<&Transaction> for zcash_primitives::transaction::Transaction {
         Ok(alt_tx)
     }
 }
+
+/// Convert a Zebra Amount into a librustzcash one.
+impl TryFrom<Amount<NonNegative>> for zcash_primitives::transaction::components::Amount {
+    type Error = ();
+
+    fn try_from(amount: Amount<NonNegative>) -> Result<Self, Self::Error> {
+        zcash_primitives::transaction::components::Amount::from_u64(amount.into())
+    }
+}
+
+/// Convert a Zebra Script into a librustzcash one.
+impl From<&Script> for zcash_primitives::legacy::Script {
+    fn from(script: &Script) -> Self {
+        zcash_primitives::legacy::Script(script.0.clone())
+    }
+}
+
+/// Compute a signature hash for a V5 transaction using librustzcash.
+///
+/// # Inputs
+///
+/// - `transaction`: the transaction whose signature hash to compute.
+/// - `hash_type`: the type of hash (SIGHASH) being used.
+/// - `input`: information about the transparent input for which this signature
+///     hash is being computed, if any. A tuple with the matching output of the
+///     previous transaction, the input itself, and the index of the input in
+///     the transaction.
+pub(crate) fn sighash_v5(
+    trans: &Transaction,
+    hash_type: &HashType,
+    input: Option<(&transparent::Output, &transparent::Input, usize)>,
+) -> blake2b_simd::Hash {
```

Comment:
I think we should just use `Hash(pub [u8; 32])` as the return type.

Otherwise, we are exposing an implementation-defined type in our API. (And the librustzcash API.)

Reply:
Sounds good, done!